extensions/coap/COAPLoader.cpp

```diff
@@ -18,7 +18,28 @@
 #include "core/FlowConfiguration.h"
 #include "COAPLoader.h"
 
+#ifdef WIN32
+#include <winsock2.h>
+#endif
+
 bool COAPObjectFactory::added = core::FlowConfiguration::add_static_func("createCOAPFactory");
+
+bool COAPObjectFactoryInitializer::initialize() {
+#ifdef WIN32
+  static WSADATA s_wsaData;
+  int iWinSockInitResult = WSAStartup(MAKEWORD(2, 2), &s_wsaData);
```

Comment:
Did you consider logging the issue in case of failed `WSAStartup`?

Reply:
`WSAStartup` is extremely unlikely to fail, that's why I didn't add one originally, but it won't hurt, so I've added it now.